src/Control/Exception/Safe.hs

```diff
@@ -172,7 +172,9 @@ evaluateDeep action = do
 catchAnyDeep :: (C.MonadCatch m, MonadIO m, NFData a) => m a -> (SomeException -> m a) -> m a
 catchAnyDeep = catchDeep
 
--- | 'catch' without async exception safety
+-- | 'catch' without async exception safety.
+--
+-- Will catch extension types which are children of 'Control.Exception.SomeAsyncException', see <https://github.com/fpco/safe-exceptions#determining-sync-vs-async>
```

Comment:
I'd recommend clarifying a bit: it will catch _all_ exception types, including `SomeAsyncException` children.

Also, two minor nits:

* s/extension/exception
* Could you wrap lines to something smaller (e.g. 80 columns)?